ui/lib/apps/ContinuousProfiling/pages/Detail.tsx

```diff
@@ -58,53 +106,99 @@ export default function Page() {
         minWidth: 150,
         maxWidth: 200,
         onRender: (record) => {
-          if (record.state === 'failed') {
-            return <Badge status="error" text={record.error} />
-          } else {
+          if (record.state === 'finished' || record.state === 'success') {
             return (
               <Badge
                 status="success"
-                text={t('continuous_profiling.detail.table.status.finished')}
+                text={t('conprof.detail.table.status.finished')}
               />
             )
           }
+          if (record.state === 'failed') {
+            return <Badge status="error" text={record.error} />
+          }
+          return <Badge text={t('conprof.list.table.status.unknown')} />
+        },
+      },
+      {
+        name: t('conprof.detail.table.columns.actions'),
+        key: 'actions',
+        minWidth: 150,
+        maxWidth: 200,
+        onRender: (record) => {
+          const rec = record as ConprofProfileDetail
+          let actionsKey = TEXT_ACTIONS
+          if (rec.profile_type !== 'goroutine') {
+            actionsKey = COMMON_ACTIONS
+          }
+          const actions = actionsKey.map((key) => ({
+            key,
+            text: t(`conprof.detail.table.actions.${key}`),
+          }))
+          return (
+            <ActionsButton
+              actions={actions}
+              disabled={rec.state !== 'finished' && rec.state !== 'success'}
+              onClick={(act) => handleClick(act, rec)}
+            />
+          )
         },
       },
     ],
     [t, profileDuration]
   )
 
-  const handleRowClick = usePersistFn(
-    async (rec, _idx, _ev: React.MouseEvent<HTMLElement>) => {
-      const {
-        profile_type,
-        target: { component, address },
-      } = rec
+  const handleClick = usePersistFn(
+    async (action: string, rec: ConprofProfileDetail) => {
+      const { profile_type, target } = rec
+      const { component, address } = target!
+      let dataFormat = ''
+      if (action === 'view_flamegraph' || action === 'download') {
+        dataFormat = 'protobuf'
+      }
       const res = await client
         .getInstance()
         .continuousProfilingActionTokenGet(
-          `ts=${ts}&profile_type=${profile_type}&component=${component}&address=${address}`
+          `ts=${ts}&profile_type=${profile_type}&component=${component}&address=${address}&data_format=${dataFormat}`
         )
       const token = res.data
       if (!token) {
         return
       }
-      window.open(
-        `${client.getBasePath()}/continuous_profiling/single_profile/view?token=${token}`,
-        '_blank'
-      )
+
+      if (action === 'view_graph' || action === 'view_text') {
+        const profileURL = `${client.getBasePath()}/continuous_profiling/single_profile/view?token=${token}`
+        window.open(profileURL, '_blank')
+        return
+      }
+
+      if (action === 'view_flamegraph') {
+        // view flamegraph by speedscope
```

Comment:
Maybe we can have a helper to opens speedscope between manual profiling and conprof.

Reply:
Yep, we can refine it later.